core/engine/src/object/builtins/jsfunction.rs

```diff
@@ -54,6 +54,12 @@ impl<A: TryIntoJsArguments, R: TryFromJs> TypedJsFunction<A, R> {
         self.inner.clone()
     }
 
+    /// Get the inner `JsFunction` without consuming this object.
+    #[must_use]
+    pub fn to_js_function(&self) -> JsFunction {
+        self.inner.clone()
+    }
```

Comment:
```suggestion
    /// Get the inner `JsFunction` without consuming this object.
    #[must_use]
    pub fn as_js_function(&self) -> &JsFunction {
        &self.inner
    }
```

The `to_` implies that the convert into a type is expensive, I think it's best to to use `as_` as suggested in the [rust api guidelines](https://rust-lang.github.io/api-guidelines/naming.html?highlight=to_#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv).

Reply:
It comes down to usage. This would be very different and much more boilerplate, as `JsFunction` isn't Copy: https://github.com/boa-dev/boa/pull/3981/files#diff-c88ba8c6fb9beadd76ba03bcc3e5dab23da271da9a9862ca2d8c565c37fdd6acR21-R27

Reply:
I kind of think we could also `impl Into<JsFunction>`, which would likely solve both. Let me try.

Reply:
Going to resolve this comment by doing the commit suggestion above, AND implementing `From<TypedJsFunction<...>> for JsFunction` if you don't need to keep the object.